llvm/lib/ExecutionEngine/JITLink/ELF_systemz.cpp

```diff
@@ -0,0 +1,440 @@
+      break;
+    }
+    case ELF::R_390_32: {
+      Kind = systemz::Pointer32;
+      break;
+    }
+    case ELF::R_390_20: {
+      Kind = systemz::Pointer20;
+      break;
+    }
+    case ELF::R_390_16: {
+      Kind = systemz::Pointer16;
+      break;
+    }
+    case ELF::R_390_12: {
+      Kind = systemz::Pointer12;
+      break;
+    }
+    case ELF::R_390_8: {
+      Kind = systemz::Pointer8;
+      break;
+    }
+    // Relocations targeting the PLT associated with the symbol.
+    case ELF::R_390_PLT64: {
+      Kind = systemz::BranchPCRelPLT64;
+      break;
+    }
+    case ELF::R_390_PLT32: {
+      Kind = systemz::BranchPCRelPLT32;
+      break;
+    }
+    case ELF::R_390_PLT32DBL: {
+      Kind = systemz::BranchPCRelPLT32dbl;
+      break;
+    }
+    case ELF::R_390_PLT24DBL: {
+      Kind = systemz::BranchPCRelPLT24dbl;
+      break;
+    }
+    case ELF::R_390_PLT16DBL: {
+      Kind = systemz::BranchPCRelPLT16dbl;
+      break;
+    }
+    case ELF::R_390_PLT12DBL: {
+      Kind = systemz::BranchPCRelPLT12dbl;
+      break;
+    }
+    case ELF::R_390_PLTOFF64: {
+      Kind = systemz::DeltaPLT64FromGOT;
+      break;
+    }
+    case ELF::R_390_PLTOFF32: {
+      Kind = systemz::DeltaPLT32FromGOT;
+      break;
+    }
+    case ELF::R_390_PLTOFF16: {
+      Kind = systemz::DeltaPLT16FromGOT;
+      break;
+    }
+    // Relocations targeting the GOT entry associated with the symbol.
+    case ELF::R_390_GOTOFF64: {
+      Kind = systemz::Delta64FromGOT;
+      break;
+    }
+    // Seems loke ‘R_390_GOTOFF32’.
+    case ELF::R_390_GOTOFF: {
+      Kind = systemz::Delta32FromGOT;
+      break;
+    }
+    case ELF::R_390_GOTOFF16: {
+      Kind = systemz::Delta16FromGOT;
+      break;
+    }
+    case ELF::R_390_GOT64: {
+      Kind = systemz::Delta64GOT;
+      break;
+    }
+    case ELF::R_390_GOT32: {
+      Kind = systemz::Delta32GOT;
+      break;
+    }
+    case ELF::R_390_GOT20: {
+      Kind = systemz::Delta20GOT;
+      break;
+    }
+    case ELF::R_390_GOT16: {
+      Kind = systemz::Delta16GOT;
+      break;
+    }
+    case ELF::R_390_GOT12: {
+      Kind = systemz::Delta12GOT;
+      break;
+    }
+    case ELF::R_390_GOTPC: {
+      Kind = systemz::DeltaPCRelGOT;
+      break;
+    }
+    case ELF::R_390_GOTPCDBL: {
+      Kind = systemz::DeltaPCRelGOTdbl;
+      break;
+    }
+    case ELF::R_390_GOTPLT64: {
+      Kind = systemz::Delta64JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTPLT32: {
+      Kind = systemz::Delta32JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTPLT20: {
+      Kind = systemz::Delta20JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTPLT16: {
+      Kind = systemz::Delta16JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTPLT12: {
+      Kind = systemz::Delta12JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTPLTENT: {
+      Kind = systemz::PCRel32JumpSlot;
+      break;
+    }
+    case ELF::R_390_GOTENT: {
+      Kind = systemz::PCRel32GOTEntry;
```

Comment:
This name doesn't really fit the others either, it should be something like `RequestGOTAndTransferToDelta32dbl`.  In the end we really shouldn't have any variant of `PCRel` anywhere, we always use `Delta` to indicate PC-relative offsets.